tests/linen/module_test.py

```diff
@@ -1371,15 +1370,18 @@ class Foo(NamedTuple):
     self.assertEqual(type(xs), Foo)  # equality test for NamedTuple doesn't check class!
 
   def test_generic_multiple_inheritance(self):
-    T = TypeVar('T')
-    class MyComponent(nn.Module, Generic[T]):
-      pass
-    class MyModule(nn.Module):
-      submodule: MyComponent[jnp.ndarray]
-    class MyComponent2(Generic[T], nn.Module):
-      pass
-    class MyModule2(nn.Module):
-      submodule: MyComponent2[jnp.ndarray]
+    if sys.version_info.major == 3 and sys.version_info.minor >= 7:
```

Comment:
So this bug only appeared with the added `__dataclass_transform__`?

Reply:
Yes because the __dataclass_transform__ only works on metaclasses so we have an empty metaclass. In the exceptional edge case where somebody wants do multiple inheritance with Generic on a Module they could probably use `six`

Reply:
Refactored the code to only add the metaclass during type checking. This avoids the problem with metaclass inheritance